crates/node/service/src/actors/rpc.rs

```diff
@@ -0,0 +1,65 @@
+//! RPC Server Actor
+
+use crate::NodeActor;
+use async_trait::async_trait;
+use jsonrpsee::server::ServerHandle;
+use tokio_util::sync::CancellationToken;
+
+/// An error returned by the [`RpcActor`].
+#[derive(Debug, thiserror::Error)]
+pub enum RpcActorError {
+    /// The [`RpcActor`]'s RPC server stopped unexpectedly.
+    #[error("RPC server stopped unexpectedly")]
+    ServerStopped,
+    /// Failed to stop the RPC server.
+    #[error("Failed to stop the RPC server")]
+    StopFailed,
+}
+
+/// An actor that handles the RPC server for the rollup node.
+#[derive(Debug)]
+pub struct RpcActor {
+    /// The handle to the RPC server.
+    server: ServerHandle,
+    /// The cancellation token, shared between all tasks.
+    cancellation: CancellationToken,
+}
+
+// impl std::fmt::Debug for RpcActor {
+//     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+//         write!(f, "RPCServerActor")
+//     }
+// }
+
+impl RpcActor {
+    /// Constructs a new [`RpcActor`] given the [`ServerHandle`] and [`CancellationToken`].
+    pub const fn new(server: ServerHandle, cancellation: CancellationToken) -> Self {
+        Self { server, cancellation }
+    }
+}
+
+#[async_trait]
+impl NodeActor for RpcActor {
+    type InboundEvent = ();
+    type Error = RpcActorError;
+
+    async fn start(mut self) -> Result<(), Self::Error> {
+        tokio::select! {
+            _ = self.server.clone().stopped() => {
+                // The server has stopped, so we should stop as well.
+                self.cancellation.cancel();
+                return Err(RpcActorError::ServerStopped);
```

Comment:
Returning an error signals to the orchestration code to cancel all other tasks. Not a problem to be explicit, but the cancellation call isn't required.

Reply:
What should the policy here be you think? Restart the server?

Reply:
For now it's fine to just halt the whole process if the server fails like you're doing now imo. Let's circle back once the node is running - I'd wager that if the server crashes, something has gone really wrong.

Reply:
But if we do see it intermittently stopping the node, totally - should just eagerly attempt to restart it such that this actor can never fail, only be cancelled.

Reply:
Ok will just add a small ticket to track for now.

I think it may make sense for periphery services like this to have some sort of restart policy by default and have a "fail fast" mode via toggle.

Reply:
https://github.com/op-rs/kona/issues/1320